File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Imports/FormContexts/Northwind.CategoryForm.cs

namespace BasicApplication.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;
    using BasicApplication;

    public partial class CategoryForm : PrefixedContext
    {
        public CategoryForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor CategoryName { get { return ById<StringEditor>("CategoryName"); } }
        public StringEditor Description { get { return ById<StringEditor>("Description"); } }
    }
}

File: common-features/src/demo.northwind/Modules/ServerTypes/Demo/Northwind.CategoryForm.ts
import { initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface CategoryForm {
    CategoryName: StringEditor;
    Description: StringEditor;
}

export class CategoryForm extends PrefixedContext {
    static readonly formKey = 'Northwind.Category';
    declare private static init: boolean;

    constructor(...args: ConstructorParameters<typeof PrefixedContext>) {
        super(...args);

        if (!CategoryForm.init) {
            CategoryForm.init = true;

            initFormType(CategoryForm, [
                'CategoryName', StringEditor,
                'Description', StringEditor
            ]);
        }
    }
}